crypto/elliptic-curve-crypto/src/public_key.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{PrivateKey, Signature};
+use zkp_elliptic_curve::{base_mul, double_base_mul, Affine, ScalarFieldElement};
+use zkp_primefield::*;
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+#[derive(PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "std", derive(Debug))]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct PublicKey(Affine);
+use crate::GENERATOR_TABLE;
+
+impl PublicKey {
+    pub fn new(x: FieldElement, y: FieldElement) -> Self {
+        Self(Affine::new(x, y))
+    }
+
+    pub fn x(&self) -> Option<&FieldElement> {
+        self.0.x()
+    }
+
+    pub fn affine(&self) -> &Affine {
```

Comment:
Should be `as_affine` if we follow the API guidelines. We can also add an `into_affine`.